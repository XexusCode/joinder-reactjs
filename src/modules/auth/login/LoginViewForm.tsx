import { Button, FormControl, FormGroup, Nav, Row } from "react-bootstrap";
import React from "react";

interface LoginViewFormProps {
  handleSubmit: (email: string, password: string) => void;
  email: string;
  handleInputEmail: (email: string) => void;
  handleInputPassword: (password: string) => void;
  password: string;
}

export const LoginViewForm = ({
  handleSubmit,
  email,
  handleInputEmail,
  handleInputPassword,
  password,
}: LoginViewFormProps) => {
  return (
    <div className="Login">
      <h1>Login Screen</h1>
      <Row>
        <form
          onSubmit={(e) => {
            e.preventDefault();
            handleSubmit(email, password);
          }}
        >
          <FormGroup controlId="email">
            <label>Email</label>
            <FormControl
              type="text"
              name="email"
              value={email}
              onChange={(e) => handleInputEmail(e.target.value)}
              placeholder="Enter your email"
            />
          </FormGroup>
          <FormGroup controlId="password">
            <label> Password </label>

            <FormControl
              type="password"
              name="password"
              value={password}
              onChange={(e) => handleInputPassword(e.target.value)}
              placeholder="Enter your password"
            />
          </FormGroup>
          <Button type="submit">Sign-In</Button>
          <Row className="mb-lg-n5">
            <Nav.Link href="/auth/register">
              ¿Aún no tienes cuenta? Regístrate
            </Nav.Link>
          </Row>
        </form>
      </Row>
    </div>
  );
};
