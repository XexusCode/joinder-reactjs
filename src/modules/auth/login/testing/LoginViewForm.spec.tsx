import * as React from "react";
import {fireEvent, render} from "@testing-library/react";
import {LoginViewForm} from "../LoginViewForm";
import {Router} from "react-router-dom";
import history from "../../../../routing/history";

describe("LoginViewForm tests", () => {
  const handleSubmit = jest.fn();
  const handleInputEmail = jest.fn();
  const handleInputPassword = jest.fn();
  jest.mock("sweetalert2", () => ({
    fire: jest.fn(),
  }));

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should make a snapshot of LoginViewForm", function () {
    const { asFragment } = render(
      <Router history={history}>
        <LoginViewForm
          handleSubmit={handleSubmit}
          email={"username"}
          handleInputEmail={handleInputEmail}
          handleInputPassword={handleInputPassword}
          password={"password"}
        />
      </Router>
    );
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });

  test("the login button calls handlesubmit", () => {
    module.exports = jest.fn();

    const {getByTestId} = render(
        <Router history={history}>
          <LoginViewForm
              handleSubmit={handleSubmit}
              email={"username"}
              handleInputEmail={() => {
              }}
              handleInputPassword={() => {
              }}
              password={"password"}
          />
        </Router>
    );
    expect(getByTestId("button-submit")).not.toHaveAttribute("disabled");
    expect(handleSubmit).not.toHaveBeenCalled();
    fireEvent.click(getByTestId("button-submit"));
    expect(handleSubmit).toHaveBeenCalled();
  });
});
