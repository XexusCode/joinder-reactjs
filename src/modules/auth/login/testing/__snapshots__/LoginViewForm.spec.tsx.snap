// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginViewForm tests should make a snapshot of LoginViewForm 1`] = `
<DocumentFragment>
  <div
    class="wrapper fadeInDown"
  >
    <div
      id="formContent"
    >
      <div
        class="fadeIn first"
      >
        <img
          alt="User Icon"
          id="icon"
          src="https://i.pinimg.com/originals/46/bf/f2/46bff2a47eea2421c50e52f1200266d5.png"
        />
        <h1>
          Joinder
        </h1>
      </div>
      <form>
        <input
          class="fadeIn second"
          data-testid="text-name"
          id="login"
          name="login"
          placeholder="Introduce tu nombre de usuario"
          type="text"
          value="username"
        />
        <input
          class="fadeIn third"
          data-testid="text-password"
          id="password"
          name="login"
          placeholder="Introduce tu contraseña"
          type="password"
          value="password"
        />
        <input
          class="fadeIn fourth"
          data-testid="button-submit"
          type="submit"
          value="Log In"
        />
      </form>
      <div
        id="formFooter"
      >
        <a
          href="/auth/register"
        >
           ¿Aún no tienes cuenta? Regístrate 
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
